tracpro/contacts/fields.py

```diff
@@ -58,3 +122,19 @@ def __init__(self, *args, **kwargs):
 
     def compress(self, values):
         return '%s:%s' % (values[0], values[1])
+
+    def clean(self, value):
+        compressed_value = super(URNField, self).clean(value)
+        scheme, address = compressed_value.split(':', 1)
+
+        if scheme == URN_SCHEME_TWITTER:
+            validate_twitter_handle(address)
+            return compressed_value
+
+        elif scheme == URN_SCHEME_TWITTERID:
+            validate_twitter_id(address)
+            return compressed_value
+
+        elif scheme == URN_SCHEME_TEL:
+            validate_phone(address)
+            return compressed_value
```

Comment:
If we get here without recognizing the scheme, we should raise a validation error.  Hopefully that won't happen (our form supposedly limits them to the schemes we know about), but say we added another scheme and forgot to update this code, we'd rather immediately get an error when we try it, than mysteriously be saving "None" as the value because that's what ``clean`` returned here.